R/ensure-format.R

```diff
@@ -90,26 +94,34 @@ ensure_format <- function(
   tab <- read_source(file_in, as_data_frame = FALSE)
 
   # write the reformatted data based on the format/ext
-  write_func <- get_write_function(format, compression)
+  write_func <- get_write_function(format, compression, num_groups)
   write_func(tab, file_out)
 
   file_out
 }
 
+get_chunk_size <- function(table, num_groups) {
+  if (is.null(num_groups)) {
+    return(NULL)
+  }
+
+  ceiling(nrow(table) / num_groups)
+}
 
 #' Get a writer
 #'
 #' @param format format to write
 #' @param compression compression to use
+#' @param num_groups how many rows groups to create (if the format supports it)
 #'
 #' @return the write function to use
 #' @export
-get_write_function <- function(format, compression) {
+get_write_function <- function(format, compression, num_groups) {
   force(compression)
   if (format == "feather") {
-    return(function(...) arrow::write_feather(..., compression = compression))
+    return(function(x, ...) arrow::write_feather(x, ..., chunk_size = get_chunk_size(x, num_groups) %||% 65536L, compression = compression))
```

Comment:
`%||% 65536L` is a bit of magic that will return `65536L` if the thing before the `%||%` is `NULL`. That number is the default for `write_feather` and we (unfortunately) can't pass `NULL` to mean default there like we can with parquet.